x/pairing/keeper/scores/score_req.go

```diff
@@ -0,0 +1,159 @@
+package scores
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	scorestypes "github.com/lavanet/lava/x/pairing/types/scores"
+	planstypes "github.com/lavanet/lava/x/plans/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+var uniformStrategy scorestypes.ScoreStrategy
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func init() {
+	uniformStrategy = make(scorestypes.ScoreStrategy)
+	uniformStrategy[scorestypes.STAKE_REQ_NAME] = 1
+}
+
+// get the overall requirements from the policy and assign slots that'll fulfil them
+// TODO: this function should be changed in the future since it only supports stake reqs
+func CalcSlots(policy planstypes.Policy) []*scorestypes.PairingSlot {
+	slots := make([]*scorestypes.PairingSlot, policy.MaxProvidersToPair)
+	stakeReq := scorestypes.StakeReq{}
+	stakeReqMap := map[string]scorestypes.ScoreReq{stakeReq.GetName(): stakeReq}
+	for i := range slots {
+		slots[i] = scorestypes.NewPairingSlot(stakeReqMap)
+	}
+
+	return slots
+}
+
+// group the slots
+func GroupSlots(slots []*scorestypes.PairingSlot) []scorestypes.PairingSlotGroup {
+	slotGroups := []scorestypes.PairingSlotGroup{}
```

Comment:
add some comments to explain what is going on through the method

Reply:
done. see https://github.com/lavanet/lava/commit/434a5174aa62741e54089e5d392dbe39d9c58d50